core/src/main/java/feign/Contract.java

```diff
@@ -249,19 +249,18 @@ protected boolean processAnnotationsOnParameter(MethodMetadata data, Annotation[
       for (Annotation annotation : annotations) {
         Class<? extends Annotation> annotationType = annotation.annotationType();
         if (annotationType == Param.class) {
-          String name = ((Param) annotation).value();
-          checkState(emptyToNull(name) != null, "Param annotation was empty on param %s.",
-              paramIndex);
+          final Param paramAnnotation = (Param) annotation;
```

Comment:
nit we don't usually mark locals final unless doing so is notable (ex used in an anonymous type)